src/timer.rs

```diff
@@ -20,6 +22,89 @@ pub struct HrTim<TIM, PSCL, CPT1, CPT2, DacRst: DacResetTrigger = NoDacTrigger>
     capture_ch2: CPT2,
 }
 
+pub struct Ch1;
+pub struct Ch2;
+
+pub trait ChExt {
+    const CH: Chan;
+}
+
+impl ChExt for Ch1 {
+    const CH: Chan = Chan::Ch1;
+}
+
+impl ChExt for Ch2 {
+    const CH: Chan = Chan::Ch2;
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum Timer {
+    Master,
+    Tim(TimX),
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum TimX {
+    A = 0,
+    B = 1,
+    C = 2,
+    D = 3,
+    E = 4,
+    #[cfg(feature = "hrtim_v2")]
+    F = 5,
+}
+
+pub trait Instance: Deref<Target = Self::RB> {
```

Comment:
Would you mind writing some very brief doc comment? For example, what is the difference of `Instance` vs `InstanceX`? :)

Reply:
So something like

```suggestion
/// Any HR Timer A-F or Master
pub trait Instance: Deref<Target = Self::RB> {
```